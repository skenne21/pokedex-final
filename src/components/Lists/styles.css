.Lists {
  
  width: 50%;
  margin: auto;
}