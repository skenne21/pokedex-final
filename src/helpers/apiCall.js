export const getTypes =  async () => {
  try {
    const response = await fetch('http://localhost:3001/types');
    const apiData = await response.json();
    return apiData;
  } catch (error) {
    throw error
  }
}

export const getPokemon = async (ids) => {
  const pokemonPromises = ids.map(async pokemon => {
    const apiData = await fetchPokemon(pokemon)
    const indivdualPokemon = {
      name: apiData.name,
      weight: apiData.weight,
      image: apiData.sprites.back_default
    }
    return indivdualPokemon;
  })  
  return Promise.all(pokemonPromises)
}

export const fetchPokemon = async (pokemon) => {
  try {
    const response = await fetch(`http://localhost:3001/pokemon/${pokemon}`);
    const apiData = await response.json();
    return apiData;
  } catch (error) {
    throw error
  }  
}


