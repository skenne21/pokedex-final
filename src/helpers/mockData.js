export const types = [
  {
    id:"1",
    name: "normal",
    pokemon: [
      '16',
      '17'
    ]
  }, {
    id: "2",
    name: "fighting",
    pokemon: [
      "56",
      "57",
      "66",
    ]
  }
]