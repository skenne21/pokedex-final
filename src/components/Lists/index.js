import React from 'react';
import './styles.css'

const Lists = ({pokemons}) => {

  const createLists = pokemons.map(pokemon => {
    return (
      <div>
        <h1>{pokemon.name}</h1>
        <img src={`${pokemon.image}`}/>
        <h2>{pokemon.weight}</h2>
      </div>
    )
  })

  return (
    <div className='Lists'>
        {createLists}
    </div>
  )
}

export default Lists;