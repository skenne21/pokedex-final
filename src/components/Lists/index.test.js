import React from 'react';
import { shallow } from 'enzyme';
import Lists from './index';
import * as mocks from '../../helpers/mockData';

describe('Lists', () => {
  let wrapper, pokemons;

  beforeEach(() => {
    pokemons = mocks.pokemons
    wrapper = shallow(
      <Lists pokemons={pokemons} />
    );
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })


});