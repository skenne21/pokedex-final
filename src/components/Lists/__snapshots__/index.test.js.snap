// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lists Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Lists
    pokemons={
      Array [
        Object {
          "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/56.png",
          "name": "mankey",
          "weight": 280,
        },
        Object {
          "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/57.png",
          "name": "primeape",
          "weight": 280,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <h1>
            mankey
          </h1>
          <img
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/56.png"
          />
          <h2>
            280
          </h2>
        </div>,
        <div>
          <h1>
            primeape
          </h1>
          <img
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/57.png"
          />
          <h2>
            280
          </h2>
        </div>,
      ],
      "className": "Lists",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "mankey0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              mankey
            </h1>,
            <img
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/56.png"
            />,
            <h2>
              280
            </h2>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "mankey",
            },
            "ref": null,
            "rendered": "mankey",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/56.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 280,
            },
            "ref": null,
            "rendered": 280,
            "type": "h2",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "primeape1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              primeape
            </h1>,
            <img
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/57.png"
            />,
            <h2>
              280
            </h2>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "primeape",
            },
            "ref": null,
            "rendered": "primeape",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/57.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 280,
            },
            "ref": null,
            "rendered": 280,
            "type": "h2",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <h1>
              mankey
            </h1>
            <img
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/56.png"
            />
            <h2>
              280
            </h2>
          </div>,
          <div>
            <h1>
              primeape
            </h1>
            <img
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/57.png"
            />
            <h2>
              280
            </h2>
          </div>,
        ],
        "className": "Lists",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "mankey0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                mankey
              </h1>,
              <img
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/56.png"
              />,
              <h2>
                280
              </h2>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "mankey",
              },
              "ref": null,
              "rendered": "mankey",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "src": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/56.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 280,
              },
              "ref": null,
              "rendered": 280,
              "type": "h2",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "primeape1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                primeape
              </h1>,
              <img
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/57.png"
              />,
              <h2>
                280
              </h2>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "primeape",
              },
              "ref": null,
              "rendered": "primeape",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "src": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/57.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 280,
              },
              "ref": null,
              "rendered": 280,
              "type": "h2",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
